src/Particular.Analyzers/Cancellation/CatchAllShouldOmitOperationCanceledAnalyzer.cs

```diff
@@ -0,0 +1,134 @@
+﻿namespace Particular.Analyzers.Cancellation
+{
+    using System.Collections.Immutable;
+    using System.Linq;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using Particular.Analyzers.Extensions;
+
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class CatchAllShouldOmitOperationCanceledAnalyzer : DiagnosticAnalyzer
+    {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(
+            DiagnosticDescriptors.CatchAllShouldOmitOperationCanceled);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+            context.RegisterCompilationStartAction(startContext =>
+            {
+                if (!(startContext.Compilation.GetTypeByMetadataName("System.Threading.CancellationToken") is INamedTypeSymbol cancellationTokenType))
+                {
+                    return;
+                }
+
+                startContext.RegisterSyntaxNodeAction(
+                    analyzeContext => Analyze(analyzeContext, cancellationTokenType),
+                    SyntaxKind.TryStatement);
+            });
+        }
+
+        static void Analyze(SyntaxNodeAnalysisContext context, INamedTypeSymbol cancellationTokenType)
+        {
+            if (!(context.Node is TryStatementSyntax tryStatement))
+            {
+                return;
+            }
+
+            foreach (var catchClause in tryStatement.ChildNodes().OfType<CatchClauseSyntax>())
+            {
+                var catchType = GetCatchType(catchClause);
+
+                if (catchType == "OperationCanceledException")
```

Comment:
I see you have this below for "System.Exception" so maybe you just missed it here?

```suggestion
                if (catchType == "OperationCanceledException" || catchType == "System.OperationCanceledException")
```

It could also be the name of a derived type (e.g. "TaskCanceledException"), an alias (a bit weird), or perhaps even a type parameter (really weird).

Ultimately, the only absolutely reliable way is to get the type from the semantic model, but I'm not sure if we want to go that far. We should never really be catching a derived type, or aliasing the type, and the type param case is just probably too weird to even think about.